src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs

```diff
@@ -54,10 +56,37 @@ public bool IsBrokerInstalledAndInvokable()
             }
         }
 
+        ///Check if the network is available. If the network is unavailable due to power optimization,
+        ///a <see cref="MsalClientException"/>will be thrown with error code <see cref="MsalError.AndroidBrokerClientPowerOptimization"/> 
+        private void CheckPowerOptimizationStatus()
+        {
+            var powerManager = PowerManager.FromContext(Application.Context);
+
+            //Power optimization checking was added in API 23
+            if (Build.VERSION.SdkInt == BuildVersionCodes.M &&
+                powerManager.IsDeviceIdleMode &&
+                !powerManager.IsIgnoringBatteryOptimizations(Application.Context.PackageName))
+            {
+                _logger.Warning("Power optimization detected on the client application: " + Application.Context.PackageName + " and the device is in doze mode or the app is standby. \n" +
+                    "Please disable power optimizations for this application to authenticate.");
+            }
+
+            var brokerPackage = _brokerHelper.Authenticators.FirstOrDefault().PackageName;
+            if (Build.VERSION.SdkInt == BuildVersionCodes.M &&
+                powerManager.IsDeviceIdleMode &&
+                !powerManager.IsIgnoringBatteryOptimizations(brokerPackage))
+            {
```

Comment:
could you combine these two? 
```csharp
if (Build.VERSION.SdkInt == BuildVersionCodes.M &&
                powerManager.IsDeviceIdleMode &&
                !powerManager.IsIgnoringBatteryOptimizations(Application.Context.PackageName) || !powerManager.IsIgnoringBatteryOptimizations(brokerPackage))
```

And then just give whichever packageName in the log message?